core/jvm/src/main/scala/scalaz/zio/RTS.scala

```diff
@@ -996,40 +894,35 @@ private object RTS {
     private final def purgeObservers(v: ExitResult[E, A], observers: List[Callback[Nothing, ExitResult[E, A]]]): Unit =
       // To preserve fair scheduling, we submit all resumptions on the thread
       // pool in order of their submission.
-      observers.reverse.foreach(k => rts.submit(k(ExitResult.Completed(v))))
+      observers.reverse.foreach(k => rts.submit(k(ExitResult.succeeded(v))))
   }
 
   sealed abstract class FiberStatus[E, A] extends Serializable with Product {
 
-    /** causes passed in when explicitly interrupting the fiber */
-    def terminationCauses: Option[List[Throwable]]
+    /** indicates if the fiber was interrupted */
+    def interrupted: Boolean
 
-    /** errors resulting from exceptions thrown during the execution of the fiber */
-    def defects: List[Throwable]
   }
   object FiberStatus extends Serializable {
     final case class Executing[E, A](
-      terminationCauses: Option[List[Throwable]],
-      defects: List[Throwable],
+      interrupted: Boolean,
```

Comment:
Am I correct to say that this reduces the memory footprint of a approximately fiber by half?

Reply:
Of the `FiberStatus`, yes (although not quite half), although the `FiberContext` has other crap (stack, etc.).